lib/process-services-cloud/src/lib/form/components/widgets/date/date-cloud.widget.ts

```diff
@@ -51,51 +49,52 @@ import { DATE_FORMAT_CLOUD } from '../../../../models/date-format-cloud.model';
     typeId = 'DateCloudWidgetComponent';
 
-    minDate: Moment;
-    maxDate: Moment;
+    minDate: string;
+    maxDate: string;
 
     private onDestroy$ = new Subject<boolean>();
 
     constructor(public formService: FormService,
-                private dateAdapter: DateAdapter<Moment>,
-                private userPreferencesService: UserPreferencesService) {
+                private dateAdapter: DateAdapter<DateFnsAdapter>,
+                private userPreferencesService: UserPreferencesService,
+                @Inject(MAT_DATE_FORMATS) private dateFormatConfig: MatDateFormats) {
         super(formService);
     }
 
     ngOnInit() {
         this.userPreferencesService
             .select(UserPreferenceValues.Locale)
             .pipe(takeUntil(this.onDestroy$))
-            .subscribe(locale => this.dateAdapter.setLocale(locale));
+            .subscribe(locale => this.dateAdapter.setLocale(DateFnsUtils.getLocaleFromString(locale)));
 
-        const momentDateAdapter = this.dateAdapter as MomentDateAdapter;
-        momentDateAdapter.overrideDisplayFormat = this.field.dateDisplayFormat;
+            this.dateFormatConfig.display.dateInput = this.field.dateDisplayFormat;
 
         if (this.field) {
             if (this.field.dynamicDateRangeSelection) {
-                const today = this.getTodaysFormattedDate();
+                const today = this.getTodaysDate();
                 if (Number.isInteger(this.field.minDateRangeValue)) {
-                    this.minDate = moment(today).subtract(this.field.minDateRangeValue, 'days');
-                    this.field.minValue = this.minDate.format(DATE_FORMAT_CLOUD);
+                    this.minDate = format(subDays(today, this.field.minDateRangeValue), DATE_FORMAT_CLOUD);
+                    this.field.minValue = this.minDate;
                 }
+
                 if (Number.isInteger(this.field.maxDateRangeValue)) {
-                    this.maxDate = moment(today).add(this.field.maxDateRangeValue, 'days');
-                    this.field.maxValue = this.maxDate.format(DATE_FORMAT_CLOUD);
+                    this.maxDate = format(addDays(today, this.field.maxDateRangeValue), DATE_FORMAT_CLOUD);
+                    this.field.maxValue = this.maxDate;
                 }
             } else {
                 if (this.field.minValue) {
-                    this.minDate = moment(this.field.minValue, DATE_FORMAT_CLOUD);
+                    this.minDate = format(new Date(this.field.minValue), DATE_FORMAT_CLOUD);
                 }
 
                 if (this.field.maxValue) {
-                    this.maxDate = moment(this.field.maxValue, DATE_FORMAT_CLOUD);
+                    this.maxDate = format(new Date(this.field.maxValue), DATE_FORMAT_CLOUD);
                 }
             }
         }
     }
 
-    getTodaysFormattedDate() {
-        return moment().format(DATE_FORMAT_CLOUD);
+    getTodaysDate() {
```

Comment:
We no longer need formatted date here as the formatter is already being used when we assign value to minDate/maxDate around line 76.

The reason why I haven't called `new Date()` directly in the code is that, In unit tests, we need to spy on `getTodaysFormattedDate()` and return a value.
Also, I have renamed `getTodaysFormattedDate()` to `getTodaysDate()` because we are not formatting the date in that function anymore.

Reply:
@SheenaMalhotra182 adding a function just to be able to create a spy for it doesn't make any sense. We shouldn't alter the component for the sake of unit tests. And you don't have to add the spy to make unit tests work in this case. You can directly call `new Date()` and then use the same in unit tests to create expected value

Reply:
changed. calling `new Date()` directly in code now